compliance/node/index/indexer.go

```diff
@@ -233,19 +243,21 @@ func runRepositoryScanner(ctx context.Context, cfg NodeIndexerConfig, l *clairco
 	return repos, nil
 }
 
-func runPackageScanner(ctx context.Context, layer *claircore.Layer) ([]*claircore.Package, error) {
-	scanner := rpm.Scanner{}
+func runPackageScanner(ctx context.Context, packageDBFilter string, layer *claircore.Layer) ([]*claircore.Package, error) {
+	scanner := rhel.PackageScanner{}
 	pkgs, err := scanner.Scan(ctx, layer)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to invoke RPM scanner")
+		return nil, errors.Wrap(err, "failed to invoke RHEL scanner")
 	}
 
 	// Filter out packages in which we are not interested.
-	// At this time, we are only interested in the RHCOS RPM database.
-	filtered := pkgs[:0]
-	for _, pkg := range pkgs {
-		if pkg.PackageDB == rhcosPackageDB {
-			filtered = append(filtered, pkg)
+	filtered := pkgs
```

Comment:
I think, we can still use the slice trick for reusing the same underlying array
```suggestion
	filtered := pkgs[:0]
```

Reply:
Won't this reset the slice length, which we do not want if `packageDBFilter == ""`?